pkg/cmd/infra/router/router.go

```diff
@@ -78,6 +78,10 @@ type RouterSelection struct {
 	// WatchEndpoints when true will watch Endpoints instead of
 	// EndpointSlices.
 	WatchEndpoints bool
+
+	// AllowExternalCertificates is set when cluster-ingress-operator propagates
+	// ROUTER_ENABLE_EXTERNAL_CERTIFICATE env var, if RouteExternalCertificate feature-gate is enabled.
```

Comment:
Updated to 
```go
// AllowExternalCertificates when true enables RouteSecretManager plugin and the external certificate validation.
// The cluster-ingress-operator sets it if RouteExternalCertificate feature-gate is enabled.
```

because it will also enable `RouteSecretManager` plugin.